### ISSUE TITLE
Dynamically node add (over addNodeMode function & directly add to dataset) issue

### ISSUE THREAD

**Issue by pablo-xor** [almende/vis#1305]
Hi,

I found small but very problematic bug during dynamically adding new nodes on specified position. In brief, my app consist of four div blocks:
```html
        <div id="content">
            <div id="tools"></div>
            <div id="status"></div>
            <div id="graph"></div>
        </div>
```
```css
#tools {
    width: 250px;
    background: #393939;
    float: left;
    position: absolute;
    z-index: 200;
    border-bottom-right-radius: 10px;
    top: 30px;
    opacity: 0.9;
}

#graph {
    width: 100%;
    z-index: 100;
    height: 100%;
    cursor: crosshair;
    position: absolute;
}

#status {
    position: absolute;
    z-index: 200;
    bottom: 10px;
    right: 10px;
    color: black;
    font-style: italic;
    text-shadow:
        -1px -1px 0 #FFF,
        1px -1px 0 #FFF,
        -1px 1px 0 #FFF,
        1px 1px 0 #FFF;  
}
```
On default viewport (canvas {x: 0.0, y: 0.0}) is OK, but when I drag viewport very much with scale >= 1.0 node will be added to dataset without refereshing canvas. I tried "redraw" method but did not work. On small scale value (0.1) is OK and when I call method "fit" after first animation frame node becomes visible.

**Comment by AlexDM0** [comment]
Hi,

Isnt the problem here that you create a node outside of the visible space? See #1235 

Regards

**Comment by pablo-xor** [comment]
Hi, 
propably not, my JS code:

```javascript
_graph.once('click', _addNodeHandler);

var _addNodeHandler = function (ctx) {
        var position = ctx.pointer.canvas;

        switch (_createTripleMode) {
            case CREATE_MODE.NODE_EDGE_NODE_START:
                _createSingleTmpNode(position, _newTripleValueA, TEMP_NEW_ID_A);
                _createTripleMode = CREATE_MODE.NODE_EDGE_NODE_END;
                _graph.once('click', _addNodeHandler);
                break;
            case CREATE_MODE.NODE_EDGE_NODE_END:
                _createSingleTmpNode(position, _newTripleValueB, TEMP_NEW_ID_B);
                _graph.addEdgeMode();
                break;
            case CREATE_MODE.NODE_EDGE:
                _createSingleTmpNode(position, _newTripleValueA, TEMP_NEW_ID_A);
                _graph.addEdgeMode();
                break;
            default:
                throw new Error('Non-handled triple creation mode!!!');
        }
    };

var _createSingleTmpNode = function (position, value, id) {
        console.log('View position:', _graph.getViewPosition());
        console.log('Created Node position:', position);
        console.log('Scale:', _graph.getScale());
        console.log('Container size: H:', _$container.height(), ' W:', _$container.width());
        
        var pos = _graph.getViewPosition(); //test
        
        _nodes.add({
            id: id,
            label: value,
            x: pos.x,
            y: pos.y
        });
    };

var setZoom = function (scale) {
        _graph.moveTo({
            position: _graph.getViewPosition(),
            scale: scale
        });

       //...
    };
```

Console output:
This works
View position: Object {x: 25.733235843029075, y: -21.593535092118827}
Created Node position: Object {x: 72.39990250969575, y: -55.34353509211883}
Scale: 1.2
Container size: H: 933  W: 1280

This not works, after call fit method is OK, but my setZoom method does not bring results.
View position: Object {x: -2694.266764156971, y: 448.4064649078812}
Created Node position: Object {x: -2832.6000974903045, y: 577.9897982412145}
Scale: 1.2
Container size: H: 933  W: 1280

But this works
View position: Object {x: -2694.2667641569697, y: 448.4064649078812}
Created Node position: Object {x: -3074.2667641569697, y: 573.4064649078812}
Scale: 0.1
Container size: H: 933  W: 1280

**Comment by AlexDM0** [comment]
Hi,

It would be very helpful if you could make a jsbin showing the problem. Would that be possible?

Regards

**Comment by pablo-xor** [comment]
https://jsbin.com/cisoro/3/edit?js,console,output

**Comment by AlexDM0** [comment]
I see, I'll see if I can fix this for the next release. Thanks for making the example!

**Comment by AlexDM0** [comment]
Hi,

Turns out this was specific to the box shape. I've push the fix for it to develop.

Thanks for notifying me!

Cheers

**Comment by AlexDM0** [comment]
Released in v4.9.0
